sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+    managed by an SRE team. In order to give more confidence to the community,
+    it is planned to transfer image building to CNCF owned accounts.
+
+#### STRIDE-SPOOF-4 Spoofing of a control plane node in a workload cluster
+
+When a worker node is created, a user could somehow interfere with the boot,
+e.g. breaking networking, assigning incorrect security groups. This would
+prevent the encrypted cloud-init data from being deleted. If the user could gain
+access to the node, they could use the IMDS provided credentials to read the
+bootstrap token from Secrets Manager and attempt to register AS a control plane
+node, since validation is only done client- side by kubeadm. This can allow the
+attacker to gain access to secrets they should not be able to.
+
+#### STRIDE-SPOOF-4 Recommended Mitigations
+
+* Replace kubeadm node join mechanism with a host attestation mechanism.
+  * Status: Planned
+
+#### STRIDE-SPOOF-5 Spoofing of a worker node in a workload cluster
+
+Any attacker controlled machine could claim to be a worker node that needs to
+join a workload cluster and request user data from Secrets Manager API.
+
+#### STRIDE-SPOOF-5 Recommended Mitigations
+
+* Same as STRIDE-SPOOF-4.
+  * Status: Planned
+
+### Tampering
+
+#### STRIDE-TAMPER-1 Tampering of Cluster API component binaries and configuration files
+
+Cluster API components (Controllers and CAPI Core) can be tampered during build
+time, during install
+
+#### STRIDE-TAMPER-1 Recommended Mitigations
+
+* At build time: Create SBoM of all the Cluster API components and verify
+  checksum as a post build action
+  *
+  Status: [Issue filed](https://github.com/kubernetes-sigs/cluster-api/issues/6153)
+* During install time: Perform mutual authentication when invoking clusterctl
+  command. Verify checksum of the downloaded components. Download with TLS
+  enabled to prevent in-transit tampering.
+  * Status: implemented / planned / to be implemented / End User Guidance
+* During runtime: Alert on modification or restarts of cluster API components to
+  detect for potential tampering.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-TAMPER-2 Tampering of Machine Images
+
+Machine images used as Node OS images, can be tampered during build time, during
+install time and during runtime
+
+#### STRIDE-TAMPER-2 Recommended Mitigations
+
+* At build time: Create SBoM of the Machine Image and verify checksum as a post
+  build action
+  *
+  Status: [Issue filed](https://github.com/kubernetes-sigs/image-builder/issues/803)
+* During install time: Perform mutual authentication when invoking clusterctl
+  command. Verify checksum of the downloaded machine image, before marking them
+  as available. Download with TLS enabled to prevent in-transit tampering.
+  * Status: Partially implemented
+    * Maintainers note: clusterctl by default uses Github APIs for releases
```

Comment:
Clusterctl doesn't download machine images. this is done by provider's controllers

Reply:
Updated to reflect this comment